model/model.go

```diff
@@ -14,18 +14,18 @@ import (
 
 // RouterConfig is the primary type used to encapsulate all router configuration.
 type RouterConfig struct {
-	WorkerProcesses          string      `json:"workerProcesses"`
-	MaxWorkerConnections     int         `json:"maxWorkerConnections"`
-	DefaultTimeout           int         `json:"defaultTimeout"`
-	ServerNameHashMaxSize    int         `json:"serverNameHashMaxSize"`
-	ServerNameHashBucketSize int         `json:"serverNameHashBucketSize"`
-	GzipConfig               *GzipConfig `json:"gzipConfig"`
-	BodySize                 int         `json:"bodySize"`
-	ProxyRealIPCIDR          string      `json:"proxyRealIpCidr"`
-	ErrorLogLevel            string      `json:"errorLogLevel"`
-	Domain                   string      `json:"domain"`
-	UseProxyProtocol         bool        `json:"useProxyProtocol"`
-	EnforceWhitelists        bool        `json:"enforceWhitelists"`
+	WorkerProcesses          string      `router:"workerProcesses"`
```

Comment:
Just talking about the struct tags, and it's mostly a paranoid thing from me. If other code decides to use a `router` struct tag, there's the conflict. And since this is an exported type, it could be used elsewhere as well.

However, given this is a router project, you're right that it's probably unlikely that there'll be a collision, so I wouldn't say it's blocking for this PR.